thorlcr/activities/csvread/thcsvrslave.cpp

```diff
@@ -99,16 +101,19 @@ class CCsvReadSlaveActivity : public CDiskReadSlaveActivityBase, public CThorDat
             localOffset = 0;
             CDiskPartHandlerBase::open();
             readFinished = false;
-            OwnedIFileIO iFileIO;
-            if (compressed)
+
             {
-                iFileIO.setown(createCompressedFileReader(iFile, activity.eexp));
-                if (!iFileIO)
-                    throw MakeActivityException(&activity, 0, "Failed to open block compressed file '%s'", filename.get());
-                checkFileCrc = false;
+                CriticalBlock block(statsCs);
```

Comment:
Yes, the gatherStats happens on another thread, and there is a tiny window where it could be invalid.